### PR TITLE
working tests inc, dec, shift, pop, push, prepend with ttl

### PR DESCRIPTION
**Pull request by kvahed**

### CODE REVIEW

File: arangod/Agency/Node.cpp [hunk 1]
```diff
@@ -495,11 +505,17 @@ ResultT<std::shared_ptr<Node>> Node::handle<INCREMENT>(VPackSlice const& slice)
...
-    } catch (std::exception const&) {
+      if (!lifetimeExpired()) {
+        tmp.add("tmp", Value(this->slice().getInt() + inc));
+        applied = true;
+      }
+    } catch (...) {}
+
+    if (!applied) {
       tmp.add("tmp", Value(1));
     }
```
Comment (goedderz) [inline arangod/Agency/Node.cpp]: If there's an exception thrown in the `try{}` part, it's not obvious to me that the builder is in a well-defined state and we can just continue adding things. I'd rather let the exception throw, unless you know something here I don't.

Comment (goedderz) [reply arangod/Agency/Node.cpp]: Same holds for the other case below.

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7736/

**Comment by KVS85** [comment]
Tests blue except known issue: https://github.com/arangodb/release-3.6/issues/15

File: arangod/Agency/Node.cpp (hunk 1 above)
Comment (joerg84) [reply arangod/Agency/Node.cpp]: @kvahed could you comment on this one?

**Comment by joerg84** [comment]
@kvahed do we have the same PR for devel?

**Pull request merged**